Content/Items/Dungeon/Weapons.ThousandthDegree.cs

```diff
@@ -0,0 +1,698 @@
+                        moveDirection.Y = -moveDirection.Y;
+                        Projectile.ai[1] = 1f;
+                    }
+                }
+                else
+                {
+                    Projectile.rotation -= (float)(moveDirection.X * moveDirection.Y) * 0.45f;
+                    if (collideX)
+                        Projectile.ai[0] = 2f;
+
+                    if (!collideX && Projectile.ai[0] == 2f)
+                    {
+                        moveDirection.Y = -moveDirection.Y;
+                        Projectile.ai[1] = 0f;
+                        Projectile.ai[0] = 1f;
+                    }
+                    if (collideY)
+                    {
+                        moveDirection.X = -moveDirection.X;
+                        Projectile.ai[1] = 0f;
+                    }
+                }
+                Projectile.velocity = speed * moveDirection;
+                Projectile.velocity = Collide();
+                if (Projectile.Center.Y < owner.Center.Y - 200 && switchDelay <= 0) //if the wheel is too far above the player it resets its movement
+                {
+                    if (!switched)
+                        switched = true;
+                    else
+                    {
+                        collided = false;
+                        Projectile.tileCollide = true;
+                    }
+
+                    switchDelay = 120;
+                }
+                Vector2 pos = Projectile.direction == 1 ? Projectile.BottomLeft : Projectile.BottomRight;
+                Dust.NewDustPerfect(pos + new Vector2(Projectile.direction * 25f, 0), ModContent.DustType<Dusts.GlowFastDecelerate>(), (Vector2.UnitX * Projectile.direction) + Vector2.UnitY * Main.rand.NextFloat(), 0, new Color(255, 160, 50), Main.rand.NextFloat(0.3f, 0.5f));
+                
+                Dust.NewDustPerfect(pos + new Vector2(Projectile.direction * 45f, 45), ModContent.DustType<Dusts.BuzzSpark>(), (Vector2.UnitX * -Projectile.direction) + Vector2.UnitY * -Main.rand.NextFloat(1f, 2f), 0, new Color(255, 160, 50), 2.3f);
+            }
+            else if (!collided)
+            {
+                Projectile.rotation += Projectile.velocity.Length() * 0.05f;
+                Projectile.velocity.Y += 0.6f;
+                if (Projectile.velocity.Y > 16f)
+                    Projectile.velocity.Y = 16f;
+
+                if (returning)
+                    collided = true;
+            }
+            else if (returning)
+            {
+                collided = true;
+                if (lerpTimer > 0)
+                {
+                    Projectile.Center = Vector2.Lerp(Projectile.Center, ((ThousandthDegreeProjectile)parentProj.ModProjectile).wheelPos, 1f - lerpTimer / 30f);
+                    lerpTimer--;
+                    if (lerpTimer <= 0)
+                        Projectile.Kill();
+                }
+                else
+                {
+                    if (Projectile.localAI[0] < 15)
```

Comment:
you should never use localAI. Always use a properly named variable or ref AI field instead.